contracts/manifold/lazyclaim/ERC1155LazyPayableClaim.sol

```diff
@@ -0,0 +1,321 @@
+            paymentReceiver: claimParameters.paymentReceiver
+        });
+    }
+
+    /**
+     * See {IERC1155LazyClaim-getClaimCount}.
+     */
+    function getClaimCount(address creatorContractAddress) external override view returns(uint256) {
+        return _claimCounts[creatorContractAddress];
+    }
+
+    /**
+     * See {IERC1155LazyClaim-getClaim}.
+     */
+    function getClaim(address creatorContractAddress, uint256 claimIndex) external override view returns(Claim memory) {
+        require(_claims[creatorContractAddress][claimIndex].storageProtocol != StorageProtocol.INVALID, "Claim not initialized");
+        return _claims[creatorContractAddress][claimIndex];
+    }
+
+    /**
+     * See {IERC1155LazyClaim-checkMintIndex}.
+     */
+    function checkMintIndex(address creatorContractAddress, uint256 claimIndex, uint32 mintIndex) public override view returns(bool) {
+        Claim storage claim = _claims[creatorContractAddress][claimIndex];
+        require(claim.storageProtocol != StorageProtocol.INVALID, "Claim not initialized");
+        require(claim.merkleRoot != "", "Can only check merkle claims");
+        uint256 claimMintIndex = mintIndex >> 8;
+        uint256 claimMintTracking = _claimMintIndices[creatorContractAddress][claimIndex][claimMintIndex];
+        uint256 mintBitmask = 1 << (mintIndex & MINT_INDEX_BITMASK);
+        return mintBitmask & claimMintTracking != 0;
+    }
+
+    /**
+     * See {IERC1155LazyClaim-checkMintIndices}.
+     */
+    function checkMintIndices(address creatorContractAddress, uint256 claimIndex, uint32[] calldata mintIndices) external override view returns(bool[] memory minted) {
+        minted = new bool[](mintIndices.length);
+        for (uint i = 0; i < mintIndices.length; i++) {
+            minted[i] = checkMintIndex(creatorContractAddress, claimIndex, mintIndices[i]);
+        }
+    }
+
+    /**
+     * See {IERC1155LazyClaim-getTotalMints}.
+     */
+    function getTotalMints(address minter, address creatorContractAddress, uint256 claimIndex) external override view returns(uint32) {
+        Claim storage claim = _claims[creatorContractAddress][claimIndex];
+        require(claim.storageProtocol != StorageProtocol.INVALID, "Claim not initialized");
+        require(claim.walletMax != 0, "Can only retrieve for non-merkle claims with walletMax");
+        return  uint32(_mintsPerWallet[creatorContractAddress][claimIndex][minter]);
+    }
+
+    /**
+     * See {IERC1155LazyClaim-mint}.
+     */
+    function mint(address creatorContractAddress, uint256 claimIndex, uint32 mintIndex, bytes32[] calldata merkleProof) external payable override {
+        Claim storage claim = _claims[creatorContractAddress][claimIndex];
+        
+        // Safely retrieve the claim
+        require(claim.storageProtocol != StorageProtocol.INVALID, "Claim not initialized");
+
+        // Check price
+        require(msg.value == claim.cost, "Must pay more.");
```

Comment:
Should this be "msg.value >= claim.cost" ?

Reply:
No, don't allow overpayment